smart-contracts/assembly/contracts/dns/__tests__/dns.spec.ts

```diff
@@ -58,6 +59,38 @@ describe('DNS contract tests', () => {
       setResolver(setResolverArgs);
     }).toThrow();
   });
+  test('valid dns entry', () => {
+    const validDnsEntries = [
+      'example',
+      'example123',
+      'example_name',
+      'example-name',
+      'example_name-123',
+    ];
+
+    validDnsEntries.forEach((entry) => {
+      expect(isDnsValid(entry)).toBe(true);
+    });
+  });
+
+  test('invalid dns entry', () => {
+    const invalidDnsEntries = [
+      'example@',
+      'example!',
+      'example$',
+      'example%',
+      'example^',
+      'example&',
+      'example*',
+      'example(',
+      'example)',
+      'example=',
+    ];
+
+    invalidDnsEntries.forEach((entry) => {
+      expect(isDnsValid(entry)).toBe(false);
+    });
```

Comment:
encapsulate these two tests in a "describe" section
descripbe("test dns name validity"
test ("valid dns name" ...
....
test("invalid dns name"....